### PR TITLE
Support for `textDocument/InlayHint`

### PR DESCRIPTION
**Pull request by angelozerr**
Support for `textDocument/InlayHint`

See https://github.com/redhat-developer/quarkus-ls/issues/595

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   "license": "Apache-2.0",
   "bugs": "https://github.com/redhat-developer/vscode-quarkus/issues",
   "engines": {
-    "vscode": "^1.37.0"
+    "vscode": "^1.65.0"
```
Comment (fbricon) [inline package.json:13]: most likely will be incompatible with theia

Comment (fbricon) [reply package.json:13]: @benoitf what's the max vscode engine version compatible with theia?

Comment (angelozerr) [reply package.json:13]: > most likely will be incompatible with theia

please note vscode-java have a PR to support Inlay Hint too, and they did that too https://github.com/redhat-developer/vscode-java/pull/2354/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R26

Comment (rgrunber) [reply package.json:13]: Yeah, it's not supported currently. https://eclipse-theia.github.io/vscode-theia-comparator/status.html .

Also, from https://github.com/redhat-developer/vscode-java/pull/2183#pullrequestreview-790383636, looks like updating the engine itself should not be harmful to Theia. Just have to ensure you don't call API that doesn't exist.

**Update** : According to https://github.com/eclipse-theia/theia/blob/master/dev-packages/application-package/src/api.ts#L21, should be 1.53.2

Comment (rgrunber) [reply package.json:13]: Also see https://github.com/redhat-developer/vscode-java/pull/1975/files for how to adopt new API while still being compatible with older engines.

**Review by AlexXuChen — APPROVED** [review]
LGTM, setting works as intended

**Comment by angelozerr** [comment]
Thanks @AlexXuChen for the review!

**Pull request merged**